app/models/results/response_tree_builder.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Results
+  # Builds a response tree from a form object
+  class ResponseTreeBuilder
+    def initialize(form)
+      @form = form
+    end
+
+    def build
+      @rn = AnswerGroup.new
+      add_children(@form.root_group.children, @rn)
+      @rn
+    end
+
+    def add_children(form_node, response_node)
```

Comment:
The var name form_node suggests it's a single node but in fact it's an array of nodes. Perhaps refactor so you're actually passing in a node and call children inside the method.